.github/workflows/main.yml

```diff
@@ -13,44 +13,45 @@ jobs:
       with:
         submodules: true
 
-    - name: Install clang (Windows)
+    - name: Install LLVM tools (Windows)
       shell: bash
       run: |
-          curl -fsSLO https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/LLVM-10.0.0-win64.exe
-          7z x LLVM-10.0.0-win64.exe -y -o"llvm"
-          echo "$(pwd)/llvm/bin" >> $GITHUB_PATH
-          echo "WASM_AR=$(pwd)/llvm/bin/llvm-ar.exe" >> $GITHUB_ENV
+        curl -fsSLO https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/LLVM-10.0.0-win64.exe
+        7z x LLVM-10.0.0-win64.exe -y -o"llvm"
+        echo "$(pwd)/llvm/bin" >> $GITHUB_PATH
+        echo "WASM_CC=$(pwd)/llvm/bin/clang.exe" >> $GITHUB_ENV
+        echo "WASM_AR=$(pwd)/llvm/bin/llvm-ar.exe" >> $GITHUB_ENV
+        echo "WASM_NM=$(pwd)/llvm/bin/llvm-nm.exe" >> $GITHUB_ENV
       if: matrix.os == 'windows-latest'
 
-    - name: Install llvm-nm (Windows)
+    - name: Override llvm-nm with one from rustup (Windows)
       run: |
         rustup update stable
         rustup default stable
         rustup component add llvm-tools-preview
-      if: matrix.os == 'windows-latest'
-
-    - name: Register llvm-nm in environment (Windows)
-      shell: bash
-      run: |
         echo "WASM_NM=$(rustc --print sysroot|sed 's|C:|/c|'|sed 's|\\|/|g')/lib/rustlib/x86_64-pc-windows-msvc/bin/llvm-nm.exe" >> $GITHUB_ENV
       if: matrix.os == 'windows-latest'
 
-    - name: Install clang (MacOS)
+    - name: Install LLVM tools (MacOS)
       shell: bash
       run: |
         curl -sSfL https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/clang+llvm-10.0.0-x86_64-apple-darwin.tar.xz | tar xJf -
         export CLANG_DIR=`pwd`/clang+llvm-10.0.0-x86_64-apple-darwin/bin
         echo "$CLANG_DIR" >> $GITHUB_PATH
         echo "WASM_CC=$CLANG_DIR/clang" >> $GITHUB_ENV
+        echo "WASM_AR=$CLANG_DIR/llvm-ar" >> $GITHUB_ENV
+        echo "WASM_NM=$CLANG_DIR/llvm-nm" >> $GITHUB_ENV
       if: matrix.os == 'macos-latest'
 
-    - name: Install clang (Linux)
+    - name: Install LLVM tools (Linux)
       shell: bash
       run: |
         curl -sSfL https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz | tar xJf -
         export CLANG_DIR=`pwd`/clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04/bin
         echo "$CLANG_DIR" >> $GITHUB_PATH
         echo "WASM_CC=$CLANG_DIR/clang" >> $GITHUB_ENV
+        echo "WASM_AR=$CLANG_DIR/llvm-ar" >> $GITHUB_ENV
+        echo "WASM_NM=$CLANG_DIR/llvm-nm" >> $GITHUB_ENV
```

Comment:
Why can't we continue to infer the second two based on the first? 

Is the plan to require all 3 going foreword?  Isn't that more/unnecessary work for somebody who wants to build wasi-libc locally?

Reply:
We can, and I don't have any intention of taking the `?= $(patsubst %clang,%llvm-nm,$(filter-out ccache sccache,$(WASM_CC)))` out of the Makefile :).

The problem is just the Github Actions provide no good way to clear the environment of whatever crap the OS dropped in there, without repeatedly doing it in every step. (`GITHUB_ENV` doesn't support `unset`, and merely extends an ambient environment.)

That is not a problem with `WASM_*` but is a problem with the standard variables, which shouldn't be initialized for native compilation when Ubuntu and friends have no idea what, if any, compilation we want to do.

So I am doing this so the last step of getting fid of the `WASM_` prefixes can be "just" a `sed`.

Reply:
I see.   Do you know if github actions actually set CC, AR, or NM?

Reply:
I am not sure, but by the names of the actions, I don't think so. I think Ubuntu and MSYS might `CC=cc`, though that clashes with the standard of what those are supposed to mean.

(Indeed, I am quite sympathetic to people making their own custom env vars in response to `CC` being set. But this to me is the moral equivalent of implementing `echo foo > file` by opening up a new file descriptor for `file` in deference to `stdout` being "always is for the PTY". `CC` is supposed to convey what C compiler is used for building things that will be installed, rather than point to the "default, native" C compiler which may or may not be suitable for that task. Both `CC` and `stdout` are parameters defined by how the parameters should used, not by what default arguments those parameters are instantiated with.)

Reply:
I did this in response to issues that the Windows CI *did* have `CC` set by default, as did the the macOS with `AR`, so I am quite sure this is not hypothetical.

If you mean, just put this commit in the same PR that strips the `WASM_` prefixes, sure. I thought it just might be useful to separate it out so that was "just a sed" and easier to review.

Reply:
I'm confused then.  I asked above "Do you know if github actions actually set CC, AR, or NM?" and your response was "I am not sure, but by the names of the actions, I don't think so.".  

But now it seems you are saying you have indeed seen evidence that the CI *is* setting these env vars?   If that is true then yes this change seem reasonable.   But lets combine it with the change to using bare CC/AR/NM.. then the history will make it clear why needed to start setting them.

Reply:
Sorry let me clear this up:

1. I know those variables are getting set.
2. I don't know whether it is an action this CI depends on or the "base image" setting them.

OK can combine.